src/main/java/org/red5/server/stream/ProviderService.java

```diff
@@ -47,8 +45,16 @@ public class ProviderService implements IProviderService {
 
 	private static final Logger log = Red5LoggerFactory.getLogger(ProviderService.class);
 
+	public static AntMediaApplicationAdapter getAppInstance(IScope scope) {
+		return (AntMediaApplicationAdapter) scope.getContext().getApplicationContext().getBean(AntMediaApplicationAdapter.BEAN_NAME);
+	}
+
 	/** {@inheritDoc} */
 	public INPUT_TYPE lookupProviderInput(IScope scope, String name, int type) {
+
+		AntMediaApplicationAdapter applicationAdapter = getAppInstance(scope);
+		applicationAdapter.fetchRtmpFromOriginIfExist(name);
+
```

Comment:
I am curious if we stop the streaming gracefully. I mean StreamFetcher in the edge nodes are being stopped gracefully when the origin stream is being stopped?